velox/exec/tests/LimitTest.cpp

```diff
@@ -63,3 +63,42 @@ TEST_F(LimitTest, basic) {
 
   assertQueryReturnsEmptyResult(makePlan(12'345, 10));
 }
+
+TEST_F(LimitTest, limitOverLocalExchange) {
+  auto data = makeRowVector(
+      {makeFlatVector<int32_t>(1'000, [](auto row) { return row; })});
+
+  auto file = TempFilePath::create();
+  writeToFile(file->path, {data});
+
+  auto planNodeIdGenerator = std::make_shared<PlanNodeIdGenerator>();
+  core::PlanNodeId scanNodeId;
+
+  CursorParameters params;
```

Comment:
I set `params.maxDrivers = 4;` and it doesn't seem to terminate. Is that expected?

Reply:
If you set params.maxDrivers, then you need to set params.numResultDrivers = 1. Would you try that?